\documentclass[12pt]{extarticle}
\usepackage[utf8]{inputenc}
\usepackage{cite}

\usepackage{xspace}
\newcommand{\ie}{\emph{i.e.}\xspace}

\title{Statistical Methods}
\date{}

\begin{document}

\maketitle

\section*{Description of the data}
We sought to characterize the dynamics of trimethylation at histone H3 at lysine 36 (H3K36me3) in a large number of genes in three biological conditions: after light activation of Set2 (writer add), and after dark inactivation of Set2 with and without the presence of Rpd3 (writer loss and writer eraser loss, respectively). H3K36me3 levels were measured through ChIP-seq across four timepoints for each condition (0/20/40/60 min for writer add and 0/30/60/90 min for the loss conditions) for 5,368 genes. Seq data was measured from three sample replicates for each condition across the timepoints. We used two different approaches to process the seq data for analysis. First, within each replicate, H3K36me3 levels for a gene and condition were scaled to the proportion of maximum H3K36me3 level observed (for said replicate, gene, and condition), resulting in a common scale across all replicates, genes, and conditions. Second, we took the measurements, previously standardized based on spike-in samples and normalized to adjust for gene length, multiplied them by 1000, and rounded to an integer, resulting in count data with a distribution proportional to the starting data. 

\section*{General statistical model}
Though the H3K36me3 data are large and well-balanced, with the vast majority of genes having complete measurements of three replicates from each condition across four timepoints, they possess challenging features from a statistical modeling perspective. Namely, the statistical model should accommodate the non-normal distribution of the H3K36me3 measurements, as well as the time course, \ie longitudinal, nature of measurements within a replicate. This broad range of features can be flexibly handled using Bayesian inference \cite{Gelman2006}. 
	The Stan statistical platform \cite{Carpenter2017} is one such computational tool for fitting complex Bayesian hierarchical models. We used the brms software package \cite{Burkner2017, Burkner2018}, which acts as a wrapper of Stan in the R statistical programming language \cite{RCoreTeam2019}. Let $y_{ijkl}$ be the H3K36me3 measurement for the $i$\textsuperscript{th} replicate sample ($i=1,2,3$) at the $j$\textsuperscript{th} timepoint ($j=0,1,2,3$) for the $k$\textsuperscript{th} gene ($k=1,2,…,5368$) for $l$\textsuperscript{th} condition ($l=1,2,3$). Briefly, we model $y$ with a GLM by defining $E(y)=g^{-1}(\eta)$, where $E(.)$ is the expected value of a random variable, $g^{-1}(.)$ is the inverse link function, and $\eta$ is the linear predictor that relates the outcome to factors of interest.
	A Bayesian analysis could in principle simultaneously model all genes and conditions, though in practice, such an approach would likely be computationally infeasible, particularly when considering the need for sufficient sampling in order for the parameter estimates to converge. To avoid these challenges, we instead model more manageable subsets of H3K36me3 data. First, we fit a model of H3K36me3 data for a specific gene and condition. Second, to make more direct comparisons between the writer loss and writer eraser loss conditions, we modeled their data jointly within a gene.
	
\section*{Zero-one-inflated Beta regression model of H3K36me3 proportions}
For the H3K36me3 data transformed to the proportion scale, we modeled the data with a zero-one-inflated beta distribution (ZOIB) and the following parameterization:
%p(y)={█(α(1- γ)@αγ@(y^(μɸ -1) 〖(1 - y)〗^((1 - μ)ɸ - 1))/(B(μɸ,   (1- μ)ɸ)))┤
 %p(y)=  (y^(μɸ -1) 〖(1 - y)〗^((1 - μ)ɸ - 1))/(B(μɸ,   (1- μ)ɸ)), where B(.) is the beta function (Casella & Berger 2002) and ɸ is a positive precision parameter. For the link function, we used the logit: g(η)="logit" (η)= "log" (η/(1- η)).





\bibliographystyle{plain}
\bibliography{stat_methods_GK}

\end{document}